File: 4031-find-all-numbers-disappeared-in-an-array-ii/4031-find-all-numbers-disappeared-in-an-array-ii.py
class Solution(object):
    def findDisappearedNumbers(self, nums, lower, upper):
        """
        :type nums: List[int]
        :type lower: int
        :type upper: int
        :rtype: List[List[int]]
        """
        li=[]
        cur=lower
        newn=sorted(set(nums))
        for i in newn:
            if lower>i:
                continue
            if upper<i:
                break
            if cur<i:
                li.append([cur,i-1])
            cur=i+1
        if (cur<=upper):
            li.append([cur,upper])
        return li

File: 4031-find-all-numbers-disappeared-in-an-array-ii/4031-find-all-numbers-disappeared-in-an-array-ii.cpp
class Solution {
public:
    vector<vector<int>> findDisappearedNumbers(vector<int>& nums, int lower, int upper) {
        vector<vector<int>> li;
        set<int> newn(nums.begin(),nums.end());
        int c=lower;
        for (int n:newn){
            if (lower>n)
                continue;
            if (upper<n)
                break;
            if (c<n)
                li.push_back({c,n-1});
            c=n+1;
        }
        if (c<=upper)
                li.push_back({c,upper});
        return li;
    }
};